src/utils/hsdis/binutils/hsdis-binutils.c

```diff
@@ -195,8 +201,11 @@ static void* decode(struct hsdis_app_data* app_data, const char* options) {
 
       int size = (*app_data->dfn)((bfd_vma) p, &app_data->dinfo);
 
-      if (size > 0)  p += size;
-      else           app_data->losing = true;
+      if (size <= 0 RISCV_ONLY(|| size == EIO)) {
```

Comment:
Reasonable, too. Will send one. Although I think we may less likely to get rid of this ungraceful check... only if all versions of binutils fix this issue.

Reply:
Sent an email yesterday and received one reply. Seems binutils should return a `-1` instead of `status`.

Not a plain text email, so there is some formatting issue. Please ignore.
https://sourceware.org/pipermail/binutils/2023-February/126188.html

RFR again for my manual tests with new hsdis installed on x86 and aarch64 JDKs passed; Hotspot tier1~4 (fastdebug) as well.